pkg/registry/rbac/clusterrole/policybased/storage.go

```diff
@@ -75,9 +76,11 @@ func (s *Storage) Create(ctx context.Context, obj runtime.Object, createValidati
 		return s.StandardStorage.Create(ctx, obj, createValidatingAdmission, options)
 	}
 
-	clusterRole := obj.(*rbac.ClusterRole)
-	rules := clusterRole.Rules
-	if err := rbacregistryvalidation.ConfirmNoEscalationInternal(ctx, s.ruleResolver, rules); err != nil {
+	clusterRole := &rbacv1.ClusterRole{}
```

Comment:
Can `rbac.ClusterRole` be used directly here?

Reply:
Corresponding decoder and gvk is required for `rbac.ClusterRole` , since there is no json tag for its fields. So we choose `rbacv1.ClusterRole` directly.